src/wallet/mod.rs

```diff
@@ -49,13 +49,36 @@ use crate::{
     cli::WalletConfig,
     convert,
     messages::{self, CoinbaseBuilder, M8_BMM_REQUEST_TAG},
+    proto::mainchain,
     types::{Ctip, SidechainAck, SidechainNumber, SidechainProposal},
     validator::Validator,
 };
 use error::WalletError;
 
 pub mod error;
 
+#[derive(Debug)]
+pub struct Deposit {
+    pub sidechain_number: u8,
+    pub address: Vec<u8>,
+    pub amount: u64,
```

Comment:
`bitcoin::amount::Amount`?

Reply:
I think u64 is better for internal stuff, `Amount` is useful for displaying and parsing Bitcoin amounts but there isn't much benefit to using it internally, it just adds more boilerplate.

Also it isn't the kind of situation where you are likely to make a type error -- you probably won't try to compute a sum of `amount` and `block_height`.